src/Microsoft.ML.Api/ApiUtils.cs

```diff
@@ -81,21 +97,58 @@ private static OpCode GetAssignmentOpCode(Type t)
             return mb.CreateDelegate(typeof(Peek<TRow, TValue>));
         }
 
+        private static Delegate GeneratePeek<TOwn, TRow, TValue>(PropertyInfo propertyInfo, OpCode assignmentOpCode)
+        {
+            // REVIEW: It seems like we really should cache these, instead of generating them per cursor.
+            Type[] args = { typeof(TOwn), typeof(TRow), typeof(long), typeof(TValue).MakeByRefType() };
+            var mb = new DynamicMethod("Peek", null, args, typeof(TOwn), true);
+            var il = mb.GetILGenerator();
+            var minfo = propertyInfo.GetGetMethod();
+            var opcode = (minfo.IsVirtual || minfo.IsAbstract) ? OpCodes.Callvirt : OpCodes.Call;
+
+            il.Emit(OpCodes.Ldarg_3);               // push arg3
+            il.Emit(OpCodes.Ldarg_1);               // push arg1
+            il.Emit(opcode, minfo);                 // push [stack top].[propertyInfo]
```

Comment:
>push [stack top].[propertyInfo] [](start = 55, length = 31)

Please reconcile the comment with the code

Reply:
I'm not quite sure what you mean here? The comment i roughly accurate, based on the similar code for fields

Reply:
I meant that the line 111 now emits a 'call' opcode, but the comment still says `push [stack top].[propertyInfo]`. Oh, but what you mean is that the result of call will be that the property value will end up on top of the stack.
I suppose it's true, but I would expect the comment to look like `// call [methodInfo]` or something.

---
In reply to: [206991828](https://github.com/dotnet/machinelearning/pull/616#discussion_r206991828) [](ancestors = 206991828)

Reply:
OK, done